blueocean-dashboard/src/main/js/creation/bitbucket/server/steps/BbChooseServerStep.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { PropTypes } from 'react';
+import { observer } from 'mobx-react';
+import { Dropdown, FormElement } from '@jenkins-cd/design-language';
+
+import FlowStep from '../../../flow2/FlowStep';
+import BbAddServerDialog from '../BbAddServerDialog';
+
+
+let t = null;
+
+
+@observer
+class BbChooseServerStep extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedServer: null,
+            showAddServerDialog: false,
+            urlErrorMsg: null,
+        };
+
+        this.dropdown = null;
+    }
+
+    componentWillMount() {
+        t = this.props.flowManager.translate;
+    }
+
+    _onChangeServerDropdown(option) {
+        const { serverManager } = this.props.flowManager;
+        serverManager.validateVersion(option.id)
+            .then(
+                success => this._onValidateVersion(success),
+                error => this._onValidateVersion(error),
```

Comment:
It is really a question of style. If you supply only a single function to `Promise.then`, it will only fire if the Promise is resolved successfully without any error. If you provide two functions then the first fires for success and the second fires for fault. `BbServerManager.validateVersion` explicitly handles both success and failure but they each `return` a value, meaning both will fire the success handler. For consistency, maybe you want to change `BbServerManager._onValidateVersionFailure` to `throw new CreateServerError` rather than return it. That's consistent with how the `_onCreateServerFailure` handler does it. If you make that change, then you want leave this code here as-is to handle both scenarios.